homeassistant/components/lock/xiaomi_aqara.py

```diff
@@ -0,0 +1,70 @@
+"""Support for Xiaomi Gateway Lock."""
+import logging
+from homeassistant.components.xiaomi_aqara import (PY_XIAOMI_GATEWAY,
+                                                   XiaomiDevice)
+from homeassistant.components.lock import LockDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+FINGER_KEY = 'fing_verified'
+PASSWORD_KEY = 'psw_verified'
+CARD_KEY = 'card_verified'
+VERIFIED_WRONG_KEY = 'verified_wrong'
+
+ATTR_VERIFIED_WRONG_TIMES = 'verified_wrong_times'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Perform the setup for Xiaomi devices."""
+    devices = []
+
+    for (_, gateway) in hass.data[PY_XIAOMI_GATEWAY].gateways.items():
+        for device in gateway.devices['lock']:
+            model = device['model']
+            if model == 'lock.aq1':
+                devices.append(XiaomiGatewayLock(device, 'Lock', gateway))
+    add_devices(devices)
+
+
+class XiaomiGatewayLock(LockDevice, XiaomiDevice):
+    """Representation of a XiaomiGatewayLock."""
+
+    def __init__(self, device, name, xiaomi_hub):
+        """Initialize the XiaomiGatewayLock."""
+        self._changed_by = 0
+        self._verified_wrong_times = 0
+
+        XiaomiDevice.__init__(self, device, name, xiaomi_hub)
+
+    @property
+    def is_locked(self) -> bool:
+        """Return true if lock is locked."""
+        return True
```

Comment:
So this entity doesn't show the current state nor can change the state of the lock. What's the idea for this platform? How should it be used?

Reply:
Xiaomi Aqara lock can not know the door's status.
This lock only can told user verified_id when someone unlock the lock.

Reply:
Is there any precedence for accepting a platform in home assistant that doesn't change state and doesn't expose any services?

Reply:
Yes. But the automation will only trigger if someone else or something else changed the lock than the previous changer. If this use case is good enough, I'm fine with this. But you should make it clear in the docs what the limitations of the platform are.

Reply:
The `last_updated` attribute will help to identify this use case.

Reply:
I have push a patch to change lock state automatically
After verity OK, the lock will be unlock state for about 5s.
Then, the lock go back unlock state automatically.